core-services/src/main/java/org/zalando/nakadi/partitioning/UserDefinedPartitionStrategy.java

```diff
@@ -10,10 +10,11 @@
 public class UserDefinedPartitionStrategy implements PartitionStrategy {
 
     @Override
-    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
+    public String calculatePartition(final EventType eventType, final PartitionData event, final List<String> partitions)
             throws PartitioningException {
         try {
-            final String partition = event.getJSONObject("metadata").getString("partition");
+            //final String partition = event.getJSONObject("metadata").getString("partition");
+            final String partition = String.valueOf(event.getPartition());
```

Comment:
`event.getPartition()` returns String

Reply:
That should be fine because we also validate whether the partition actually exists and the Topic Repository sends back the list of partitions in String[] either way.